library/posture.c

```diff
@@ -236,39 +238,36 @@ void ziti_send_posture_data(ziti_context ztx) {
                         newProcInfo->query = query;
                         newProcInfo->service = service;
                         model_map_set(&processes, query->process->path, newProcInfo);
-                        if (procInfo == NULL) {
-                            procInfo = newProcInfo;
-                        }
                     }
-                } else if (strcmp(query->query_type, PC_PROCESS_MULTI_TYPE) == 0){
+
+                } else if (strcmp(query->query_type, PC_PROCESS_MULTI_TYPE) == 0) {
                     int processIdx = 0;
-                    while(query->processes[processIdx] != NULL) {
-                        ziti_process* process = query->processes[processIdx];
+                    while (query->processes[processIdx] != NULL) {
+                        ziti_process *process = query->processes[processIdx];
 
                         void *curVal = model_map_get(&processes, process->path);
                         if (curVal == NULL) {
                             NEWP(newProcInfo, struct query_info);
                             newProcInfo->query_set = set;
                             newProcInfo->query = query;
                             newProcInfo->service = service;
-                            model_map_set(&processes, process->path, newProcInfo);
-                            if (procInfo == NULL) {
-                                procInfo = newProcInfo;
-                            }
+
+                            model_map_set(&processes, query->process->path, newProcInfo);
                         }
                         processIdx++;
                     }
                 }
                 queryIdx++;
             }
-            setIdx++;
         }
     }
 
-    // mark all responses obsolete in case they were removed
+    // mark responses obsolete in case they were removed
     pr_info *resp;
     MODEL_MAP_FOREACH(name, resp, &ztx->posture_checks->responses) {
-        resp->obsolete = true;
+        if (!resp->pending && !resp->should_send) {
```

Comment:
This was marking process responses as obsolete before they were processed on the worker. The worker would finish and the process response would not be sent as it was marked obsolete.